### ISSUE TITLE
Reverse-engineering the original app

### ISSUE THREAD

**Issue by Huy-Ngo** [Huy-Ngo/PolyGlotAndroid#1]
Before starting, we need to understand the workflow of the app first. To do that, we read the source code of PolyGlot to understand the classes and how they interact. Please summarize it in an understandable way and reasonably short.

**Comment by DraqueT** [comment]
Here is the flow when a file is loaded into PolyGlot. PolyGlot files are simply zip archives, which you can open up to see the structure of. (If you need me to go over their anatomy, make another ticket, and I will create some documentation there):

- a DictCore object is instantiated
- readFile in the DictCore is called with a path to the PolyGlot language file
    - IOHandler.loadImageAssets is called, loading all saved images into the imageCollection object of the DictCore
    - PFontHandler.setFontFrom is called, fetching fonts from the archive (local fonts are currently disabled, but can be reenabled by removing the confont check from the initial lines of PFontHandler.setFontFrom())
    - Using a CustHandler, XML representing the language is parsed and fed back into the DictCore. Do not worry about overrideXML for now. This had to do with a backup restoration feature.
    - Logographic data is loaded (currently disabled, but I can make a PR that re-enables it to your project once you are ready)
    - Reversion states are loaded (this has to do with the backup restoration, don't worry about it for now
   - if errors or warnings came up during load, they are displayed to the user

Please let me know if this is enough for you to work from or if you need additional documentation. If we need to, we can do a facetime and actually talk to each other. That might be a faster way to communicate.

**Comment by Huy-Ngo** [comment]
Thank you. I really appreciate your participation. I'm currently stuck with two other projects so I'm working on this really slowly.
